R/fda-table_05.R

```diff
@@ -0,0 +1,81 @@
+# Pre-processing
+
+library(dplyr)
+library(scda)
+library(tern)
+library(lubridate)
+
+adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl") %>%
+  filter(SAFFL == "Y")
+
+adex <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adex") %>%
+  filter(SAFFL == "Y") %>%
+  select("USUBJID", "ARM", "TRTSDTM", "TRTEDTM") %>%
+  distinct() %>%
+  mutate(TRTS_D = as.Date(ymd_hms(TRTSDTM))) %>%
+  mutate(TRTE_D = as.Date(ymd_hms(TRTEDTM))) %>%
+  mutate(TRTDUR = as.numeric(TRTE_D - TRTS_D + 1)) %>%
+  mutate(
+    any_d = (TRTDUR > 0),
+    d_less_1 = (TRTDUR < 365 / 12 * 1),
+    d_more_1 = (TRTDUR >= 365 / 12 * 1),
+    d_more_3 = (TRTDUR >= 365 / 12 * 3),
+    d_more_6 = (TRTDUR >= 365 / 12 * 6),
+    d_more_12 = (TRTDUR >= 365 / 12 * 12)
+  ) %>%
+  formatters::var_relabel(
+    any_d = "Any duration (at least 1 dose)",
+    d_less_1 = "<1 month",
+    d_more_1 = ">=1 month",
+    d_more_3 = ">=3 months",
+    d_more_6 = ">=6 months",
+    d_more_12 = ">=12 months"
+  )
+
+# Build layout
+
+lyt <- basic_table(show_colcounts = TRUE) %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "TRTDUR", var_labels = "Duration of treatment, weeks (or months, or days, or cycles)", function(x, ...) {
+    if (is.numeric(x)) {
+      in_rows(
+        "Mean (sd)" = c(mean(x), sd(x)),
+        "Median (min - max)" = c(median(x), range(x)),
+        "Interquartile range" = c(quantile(x, 0.25), quantile(x, 0.75)),
+        "Total exposure (person years)" = c(sum(x), sum(x) / 365),
+        .formats = c("xx.xx (xx.xx)", "xx.xx (xx.xx - xx.xx)", "xx.xx - xx.xx", "xx.xx (xx.xx)")
+      )
+    } else {
+      stop("type not supported")
+    }
+  }) %>%
+  analyze("any_d", var_labels = "Patients treated, by duration, n (%)", function(x, .N_col) {
```

Comment:
I think we can exclude the format string here since it is implied (and removed everywhere else in the table)
```suggestion
  analyze("any_d", var_labels = "Patients treated, by duration", function(x, .N_col) {
```